### PR TITLE
Add Memfault integration

### PR DESCRIPTION
**Pull request by michal-narajowski**
https://memfault.com/

### CODE REVIEW

File: sys/memfault/src/memfault_platform_core.c [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    metrics_period_sec = period_sec;
+    metrics_callback = callback;
+    metrics_callout_cb(NULL);
+    return true;
+}
+
+uint64_t
+memfault_platform_get_time_since_boot_ms(void)
+{
+    return (uint64_t) (os_get_uptime_usec() / 1000);
```
Comment (kasjer) [inline sys/memfault/src/memfault_platform_core.c:89]: case here seems superfluous

File: sys/memfault/src/memfault_platform_core.c [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+{
+    static uint8_t s_event_storage[MYNEWT_VAL(MEMFAULT_EVENT_STORAGE_SIZE)];
+    int rc;
+
+    SYSINIT_ASSERT_ACTIVE();
+    os_callout_init(&metrics_callout, os_eventq_dflt_get(),
+                    metrics_callout_cb, NULL);
+    SYSINIT_PANIC_ASSERT(metrics_callout.c_evq != NULL);
+
+    const sResetBootupInfo reset_reason = {
```
Comment (kasjer) [inline sys/memfault/src/memfault_platform_core.c:103]: variables are declared all over the place in this function

File: sys/memfault/pkg.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+pkg.deps:
+    - "@apache-mynewt-core/kernel/os"
+    - "@apache-mynewt-core/sys/shell"
+    - "@apache-mynewt-core/sys/flash_map"
+    - "@apache-mynewt-core/hw/hal"
+    - "@apache-mynewt-core/mgmt/imgmgr"
+    - "@mcuboot/boot/bootutil"
+
+pkg.deps.MEMFAULT_MGMT:
+    - "@apache-mynewt-mcumgr/cmd/memfault_mgmt"
```
Comment (kasjer) [inline sys/memfault/pkg.yml:35]: non existing package

**Comment by michal-narajowski** [comment]
Depends on these pull requests:
https://github.com/apache/mynewt-mcumgr/pull/84
https://github.com/apache/mynewt-newtmgr/pull/165

**Comment by michal-narajowski** [comment]
Closing since this feature has been added to memfault repo as a newt package: https://github.com/memfault/memfault-firmware-sdk/tree/master/ports/mynewt

**Pull request closed without merging**